scripts/split_repo_migration/git-migrate-history.sh

```diff
@@ -0,0 +1,167 @@
+# "googleapis/python-speech" \
+# "~/temp/google-cloud-python" \
+# "" \
+# "" \
+# ".github/.OwlBot.yaml"
+#
+# This invocation runs migration script on python-speech, where the source path is root directory,
+# nothing is ignored, and .OwlBot.yaml is being kept.
+
+set -ex
+
+[[ "${BASH_SOURCE[0]}" != "${0}" ]] && EXIT=return || EXIT=exit
+
+if [ $# -lt 3 ]
+then
+  echo "Usage: $0 <source-repo> <target-repo> <source-path> [folders,to,skip] [files,to,keep] [branch-name]"
+  $EXIT 1
+fi
+
+# source GitHub repository. format: <owner>/<repo>
+SOURCE_REPO=$1
+
+# destination GitHub repository. format: <owner>/<repo>
+TARGET_REPO=$2  # Path to google-cloud-python
+
+# path in the source repo to copy code from. Defaults to the root directory
+SOURCE_PATH=$3
+
+# comma-separated list of files/folders to skip
+IGNORE_FOLDERS=$4
+
+# keep these specific files that would otherwise be deleted by IGNORE_FOLDERS
+KEEP_FILES=$5
+
+# override the HEAD branch name for the migration PR
+BRANCH=$6
+
+
+if [[ -z "${BRANCH}" ]]
+then
+  # default the branch name to be generated from the source repo name
+  BRANCH=$(basename ${SOURCE_REPO})-migration
+fi
+export FILTER_BRANCH_SQUELCH_WARNING=1
+
+# create a working directory
+WORKDIR="$(mktemp -d -t code-migration-XXXXXXXXXX)"
+echo "Created working directory: ${WORKDIR}"
+
+pushd "${WORKDIR}"  # cd into workdir
+
+echo "Cloning source repository: ${SOURCE_REPO}"
+git clone "git@github.com:${SOURCE_REPO}.git" source-repo
+
+pushd source-repo
+git remote remove origin
+
+# prune only files within the specified directory
+if [[ ! -z "${SOURCE_PATH}" ]]
+then
+  echo "Pruning commits only including path: ${SOURCE_PATH}"
+  git filter-branch \  # rewrites history... unsafe
+    --prune-empty \
+    --subdirectory-filter "${SOURCE_PATH}"
+fi
+
+if [[ ! -z "${IGNORE_FOLDERS}" ]]
+then
+  echo "Ignoring folder: ${IGNORE_FOLDERS}"
+  mkdir -p "${WORKDIR}/filtered-source"
+  FOLDERS=$(echo ${IGNORE_FOLDERS} | tr "," " ")
+  # remove files/folders we don't want
+  FILTER="(rm -rf ${FOLDERS} || true)"
+  if [[ ! -z "${KEEP_FILES}" ]]
+  then
+    KEEP_FILES_SPACES=($(echo ${KEEP_FILES} | tr "," " "))
+    LAST_ELEMENT=$(( ${#KEEP_FILES_SPACES[@]} - 1 ))
+    KEEP_FILE_COMMANDS=""
+    for file in "${KEEP_FILES_SPACES[@]}"
+    do
+      if [[ $file == "${KEEP_FILES_SPACES[$LAST_ELEMENT]}" ]]
+      then
+        KEEP_FILE_COMMANDS+="git checkout -- ${file} 2>/dev/null || true"
+      else 
+        KEEP_FILE_COMMANDS+="git checkout -- ${file} 2>/dev/null || true; "
+      fi   
+    done
+    # restore files to keep, silence errors if the file doesn't exist
+    FILTER="${FILTER}; ${KEEP_FILE_COMMANDS}"
+  fi
+  git filter-branch \
+    --force \
+    --prune-empty \
+    --tree-filter "${FILTER}"
+fi
+
+TARGET_PATH="packages/$(jq -r '.distribution_name' .repo-metadata.json)"  # -r removes quotes around the name.
+
+# reorganize the filtered code into the desired target locations
+if [[ ! -z "${TARGET_PATH}" ]]
```

Comment:
This will always evaluate to true, so we should just take out the conditional. Or we can have the conditional on just the output of `jq`.